sharing_portal/views.py

```diff
@@ -355,11 +409,262 @@ def launch(request, artifact, artifact_versions, version_idx=None):
+        day_pass_request_url = preserve_sharing_key(
+            reverse("sharing_portal:request_day_pass", args=[artifact.pk]), request
+        )
+        return redirect(day_pass_request_url)
+
+    version.launch_count = F("launch_count") + 1
+    version.save(update_fields=["launch_count"])
     return redirect(version.launch_url(can_edit=can_edit(request, artifact)))
 
 
+@check_view_permission
+@login_required
+def request_day_pass(request, artifact, **kwargs):
+    if not artifact or not artifact.is_reproducible:
+        raise Http404("That artifact either doesn't exist, or can't be reproduced")
+
+    if request.method == "POST":
+        form = RequestDayPassForm(
+            request.POST,
+            request,
+        )
+        if form.is_valid():
+            day_pass_request = DayPassRequest.objects.create(
+                artifact=artifact,
+                name=form.cleaned_data["name"],
+                institution=form.cleaned_data["institution"],
+                reason=form.cleaned_data["reason"],
+                created_by=request.user,
+                status=DayPassRequest.STATUS_PENDING,
+            )
+            send_request_mail(day_pass_request, request)
+
+            messages.add_message(request, messages.SUCCESS, "Request submitted")
+            return HttpResponseRedirect(
+                preserve_sharing_key(
+                    reverse("sharing_portal:detail", args=[artifact.pk]), request
+                )
+            )
+        else:
+            if form.errors:
+                (messages.add_message(request, messages.ERROR, e) for e in form.errors)
+            return HttpResponseRedirect(
+                preserve_sharing_key(
+                    reverse("sharing_portal:request_day_pass", args=[artifact.pk]),
+                    request,
+                )
+            )
+
+    form = RequestDayPassForm(
+        initial={
+            "name": f"{request.user.first_name} {request.user.last_name}",
+            "email": request.user.email,
+        }
+    )
+
+    template = loader.get_template("sharing_portal/request_day_pass.html")
+    context = {
+        "artifact": artifact,
+        "form": form,
+    }
+    return HttpResponse(template.render(context, request))
+
+
+def send_request_mail(day_pass_request, request):
+    url = request.build_absolute_uri(
+        reverse("sharing_portal:review_day_pass", args=[day_pass_request.id])
+    )
+    help_url = request.build_absolute_uri(reverse("djangoRT:mytickets"))
+    list_url = request.build_absolute_uri(
+        reverse("sharing_portal:list_day_pass_requests")
+    )
+    subject = f'Day pass request for "{day_pass_request.artifact.title}"'
+    body = f"""
+    <p>
+    A request has been made to reproduce the artifact:
+    {day_pass_request.artifact.title}.
+    </p>
+    <p>
+    By approving this day pass, the requesting user will have access to
+    Chameleon under your oversight, with the intention of reproducing your
```

Comment:
Something to tweak later... it may make sense to make the mail pretty innocuous and put this information in the review page instead. Otherwise we are maintaining a lot of text in various places.

Reply:
I've removed this in favor of text on the review page.